### PR TITLE
pin-1858: Token generation rate limiting

### PR DESCRIPTION
**Pull request by galales**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/authorizationserver/api/impl/AuthApiServiceImpl.scala [hunk 1]
```diff
@@ -76,6 +80,7 @@ final case class AuthApiServiceImpl(
       clientUUID    <- checker.subject.toFutureUUID
       keyWithClient <- getTokenGenerationBundle(clientUUID, checker.kid)
       _             <- verifyClientAssertion(keyWithClient, checker)
+      _             <- rateLimiter.rateLimiting(keyWithClient.client.consumerId)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/authorizationserver/api/impl/AuthApiServiceImpl.scala]: here most of the work has already been done, so this rate limiting does not actually prevent loading our systems, but unfortunately this check cannot be performed earlier

Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/authorizationserver/api/impl/AuthApiServiceImpl.scala]: Could be useful to work also on IP?

Comment (galales) [reply src/main/scala/it/pagopa/interop/authorizationserver/api/impl/AuthApiServiceImpl.scala]: that job should be already done by AWS WAF

**Review by beetlecrunch — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/authorizationserver/api/impl/AuthApiServiceImpl.scala (hunk 1 above)
Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/authorizationserver/api/impl/AuthApiServiceImpl.scala]: Will this prevent DOS but not DDOS 🤔 ?

**Review by TonioGela — APPROVED** [review]

**Pull request merged**